@nstosic I'm afraid I can't get this to build - did you?

To begin I get the error noted in my PR re: gradle version & shared preferences.
```
A problem occurred evaluating root project 'shared_preferences'.
> Failed to apply plugin [id 'com.android.library']
   > Minimum supported Gradle version is 5.1.1. Current version is 4.10.2. If using the gradle wrapper, try editing the distributionUrl in /home/nigel/.pub-cache/hosted/pub.dartlang.org/shared_preferences-0.5.4+3/android/gradle/wrapper/gradle-wrapper.properties to gradle-5.1.1-all.zip
```
My branch `cloudwheels:cloudwheels-fixes` will build successfully after making the update to 5.1.1 in gradle-wrapper.properties. With this branch however I get:

```
...
FAILURE: Build failed with an exception.                                                                           
                                                                                                                   
* What went wrong:                                                                                                 
Execution failed for task ':app:processDebugManifest'.                                                             
> Manifest merger failed : Attribute application@appComponentFactory value=(androidx.core.app.CoreComponentFactory) from [androidx.core:core:1.0.2] AndroidManifest.xml:22:18-86
        is also present at [com.android.support:support-compat:28.0.0] AndroidManifest.xml:22:18-91 value=(android.support.v4.app.CoreComponentFactory).
        Suggestion: add 'tools:replace="android:appComponentFactory"' to <application> element at AndroidManifest.xml:15:5-59:19 to override.
...
```

If I make the suggested change to the manifest  (also adding the namespace declaration in the header for tools to avoid another error i.e. `xmlns:tools="http://schemas.android.com/tools"`) then I get the following error:
```
...
/home/nigel/toptal-fixes/android/app/src/main/AndroidManifest.xml:16:5-61:19 Error:
        tools:replace specified at line:16 for attribute android:appComponentFactory, but no new value specified
/home/nigel/toptal-fixes/android/app/src/main/AndroidManifest.xml Error:
        Validation failed, exiting                                      
                                                                        
FAILURE: Build failed with an exception.                                
                                                                        
* What went wrong:                                                      
Execution failed for task ':app:processDebugManifest'.                  
> Manifest merger failed with multiple errors, see logs  
...
```

Unfortnately I am out of time to investigate further right now.

I am just about to update flutter version but current system is:
```
Flutter (Channel master, v1.10.15-pre.186, on Linux, locale en_US.UTF-8)
    • Flutter version 1.10.15-pre.186 at /home/nigel/flutter
    • Framework revision 1c535de7e0 (12 days ago), 2019-10-18 16:40:42 -0700
    • Engine revision 8aefcd8575
    • Dart version 2.6.0 (build 2.6.0-dev.8.0 a61c775db8)
```

--- a/README.md
+++ b/README.md
@@ -4,6 +4,10 @@
 
 Cross-platform demo chat client developed using Flutter & Firebase for Toptal Engineering Blog article.
 
+# Build environment
+
+This project is developed to work with `flutter channel stable`. There is no guarantee that it will work on different flutter channels.
+
 # API keys
 
 Note - if you clone this repository and try running the project, it'll fail because I've removed API keys for Facebook and Firebase. Refer to the [Facebook](https://developers.facebook.com/docs/facebook-login/) or [Firebase](https://firebase.google.com/docs/flutter/setup) official documentation for a step-by-step guide to setting up the project.

--- a/android/gradle.properties
+++ b/android/gradle.properties
@@ -1,3 +1,4 @@
 org.gradle.jvmargs=-Xmx1536M
 android.useAndroidX=true
 android.enableJetifier=true
+android.enableR8=true

--- a/android/gradle/wrapper/gradle-wrapper.properties
+++ b/android/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip

